### PR TITLE
Query pipelines BDD

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## Usage and product changes
BDD covering common cases for query pipelines

## Implementation
BDD covering common cases for query pipelines

### CODE REVIEW

File: query/language/pipelines.feature [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    """
+    match
+      $p isa person, has name "Alice";
+    select $p;
+    """
+    Then uniquely identify answer concepts
+      | p              |
+      | key:ref:0      |
+
+  Scenario: A match can be chained with a delete and insert to emulate an update.
```
Comment (flyingsilverfin) [inline query/language/pipelines.feature:104]: Consistent spacing between scenarios please, check what we have elsewhere

File: query/language/pipelines.feature [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    """
+    Then uniquely identify answer concepts
+      | p              |
+      | key:ref:0      |
+
+    Given get answers of typeql read query
+    """
+    match
+      $p isa person, has name $name;
+    sort $name desc;
```
Comment (flyingsilverfin) [inline query/language/pipelines.feature:161]: Maybe you should extend this with one that sorts on name, then something else? The test says "set of variables"?

File: query/language/pipelines.feature [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      | age             |
+      | attr:age:2      |
+
+    Given get answers of typeql read query
+    """
+    match
+      $p1 isa person, has age $age;
+    sort $age desc;
+    offset 2;
+    limit 2;
```
Comment (flyingsilverfin) [inline query/language/pipelines.feature:197]: Can you add one that chains them longer and in different order too?
```
limit
sort
select
offset
sort
limit
```

File: query/language/pipelines.feature (hunk 2 above)
Comment (krishnangovindraj) [reply query/language/pipelines.feature:163]: Renamed and moved the multi-variable case below

File: query/language/pipelines.feature (hunk 3 above)
Comment (krishnangovindraj) [reply query/language/pipelines.feature:190]: Did something smaller

**Pull request merged**